addons/website_sale_stock/tests/test_website_sale_stock_product_template.py

```diff
@@ -91,3 +91,29 @@ def test_get_additional_combination_info_max_combo_quantity_without_max(self):
             )
 
         self.assertNotIn('max_combo_quantity', combination_info)
+
+    def test_get_additional_combination_info_free_quantity_is_integer(self):
+        """
+        Ensure that free quantity returned in website_sale_stock is always an
+        integer. Previously, with 9 units in stock and a pack of 6 UoM, the
+        computed free quantity was 1.5 instead of 1.
+        """
+        product = self._create_product(is_storable=True, allow_out_of_stock_order=False)
+        self.env['stock.quant'].create([
+            {
+                'product_id': product.id,
+                'location_id': self.warehouse.lot_stock_id.id,
+                'quantity': 9,
+            },
+        ])
```

Comment:
Stock side we usually call `self.env['stock.quant']._update_available_quantity`
Better if a quant already exists and in case of refactoring :)